diff --git a/nmake.opt b/nmake.opt
index 936cfcc..bf7072e 100644
--- a/nmake.opt
+++ b/nmake.opt
@@ -215,7 +215,7 @@ INCLUDE_OGR_FRMTS = YES
 BSB_SUPPORTED = 1

 # Comment out the following to disable ODBC support.
-ODBC_SUPPORTED = 1
+#ODBC_SUPPORTED = 0

 # Comment out the following to disable JPEG support.
 JPEG_SUPPORTED = 1
@@ -224,9 +224,9 @@ JPEG_SUPPORTED = 1
 JPEG12_SUPPORTED = 1

 #if using an external jpeg library uncomment the follwing lines
-#JPEG_EXTERNAL_LIB = 1
+JPEG_EXTERNAL_LIB = 1
 #JPEGDIR = c:/projects/jpeg-6b
-#JPEG_LIB = $(JPEGDIR)/libjpeg.lib
+JPEG_LIB = $(JPEGDIR)/build/sharedlib/Release/jpeg.lib

 #if using an external png library uncomment the follwing lines
 #PNG_EXTERNAL_LIB = 1
@@ -318,9 +318,8 @@ JPEG12_SUPPORTED = 1
 #		$(OGDIDIR)\lib\$(TARGET)\zlib_ogdi$(OGDIVER).lib

 # Uncomment for Expat support (required for KML, GPX and GeoRSS read support).
-#EXPAT_DIR = "C:\Program Files\Expat 2.0.1"
-#EXPAT_INCLUDE = -I$(EXPAT_DIR)/source/lib
-#EXPAT_LIB = $(EXPAT_DIR)/bin/libexpat.lib
+EXPAT_INCLUDE = -I$(EXPAT_DIR)/lib
+EXPAT_LIB = $(EXPAT_DIR)/win32/bin/Release/libexpat.lib

 # Uncomment for Xerces based GML and ILI support.
 #XERCES_DIR =	c:\warmerda\supportlibs\xerces-c_2_6_0
@@ -626,7 +626,7 @@ GDALLIB	=    $(GDAL_ROOT)/gdal.lib
 !ENDIF

 !IFDEF ODBC_SUPPORTED
-ODBCLIB = odbc32.lib odbccp32.lib user32.lib
+ODBCLIB = legacy_stdio_definitions.lib odbc32.lib odbccp32.lib user32.lib
 !ENDIF

 !IF DEFINED(MRSID_DIR) || DEFINED(MRSID_RASTER_DIR) || DEFINED(MRSID_LIDAR_DIR)
diff --git a/port/cpl_config.h.vc b/port/cpl_config.h.vc
index bfff8f6..35311b2 100644
--- a/port/cpl_config.h.vc
+++ b/port/cpl_config.h.vc
@@ -17,7 +17,9 @@
 #if defined(_MSC_VER) && (_MSC_VER < 1500)
 #  define vsnprintf _vsnprintf
 #endif
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
 #define snprintf _snprintf
+#endif

 #define HAVE_GETCWD 1
 /* gmt_notunix.h from GMT project also redefines getcwd. See #3138 */
