diff --git a/configure b/configure
index d2998c8..b4c5e1c 100755
--- a/configure
+++ b/configure
@@ -655,7 +655,7 @@ def configure_winsdk(o):
 
   winsdk_dir = os.environ.get("WindowsSdkDir")
 
-  if winsdk_dir and os.path.isfile(winsdk_dir + '\\bin\\ctrpp.exe'):
+  if winsdk_dir and os.path.isfile(winsdk_dir + '\\bin\\x64\\ctrpp.exe'):
     print "Found ctrpp in WinSDK--will build generated files into tools/msvs/genfiles."
     o['variables']['node_has_winsdk'] = 'true'
     return
@@ -724,7 +724,7 @@ if options.use_ninja:
 elif options.use_xcode:
   gyp_args += ['-f', 'xcode']
 elif flavor == 'win':
-  gyp_args += ['-f', 'msvs', '-G', 'msvs_version=auto']
+  gyp_args += ['-f', 'msvs', '-G', 'msvs_version=2013']
 else:
   gyp_args += ['-f', 'make-' + flavor]
 
