diff --git a/Makefile.vc b/Makefile.vc
index 8135bc1..648fe62 100644
--- a/Makefile.vc
+++ b/Makefile.vc
@@ -25,7 +25,7 @@ PLATFORM_LDFLAGS = /SAFESEH
 ## Nothing more to do below this line!
 
 NOLOGO     = /nologo
-CCNODBG    = cl.exe $(NOLOGO) /O2 /DNDEBUG
+CCNODBG    = cl.exe $(NOLOGO) /O2 /DNDEBUG /Zi /DEBUG
 CCDEBUG    = cl.exe $(NOLOGO) /Od /Gm /Zi /D_DEBUG /RTC1
 CFLAGS     = /Isrc $(NOLOGO) /W3 /EHsc /c /GS
 CFLAGS     = $(CFLAGS) /DWIN32 /D_CRT_SECURE_NO_WARNINGS /DWIN32_LEAN_AND_MEAN
