diff --git a/contrib/vstudio/vc11/testzlib.vcxproj b/contrib/vstudio/vc11/testzlib.vcxproj
index 0115dd1..3ea06ec 100644
--- a/contrib/vstudio/vc11/testzlib.vcxproj
+++ b/contrib/vstudio/vc11/testzlib.vcxproj
@@ -209,6 +209,7 @@
       <DataExecutionPrevention>
       </DataExecutionPrevention>
       <TargetMachine>MachineX86</TargetMachine>
+      <ImageHasSafeExceptionHandlers>false</ImageHasSafeExceptionHandlers>
     </Link>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='ReleaseWithoutAsm|Win32'">
@@ -270,6 +271,7 @@
       <DataExecutionPrevention>
       </DataExecutionPrevention>
       <TargetMachine>MachineX86</TargetMachine>
+      <ImageHasSafeExceptionHandlers>false</ImageHasSafeExceptionHandlers>
     </Link>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
diff --git a/contrib/vstudio/vc11/zlibvc.def b/contrib/vstudio/vc11/zlibvc.def
index 980fed3..ca2c8ec 100644
--- a/contrib/vstudio/vc11/zlibvc.def
+++ b/contrib/vstudio/vc11/zlibvc.def
@@ -1,7 +1,7 @@
 LIBRARY
 ; zlib data compression and ZIP file I/O library
 
-VERSION		1.2.8
+;VERSION		1.2.8
 
 EXPORTS
         adler32                                  @1
diff --git a/contrib/vstudio/vc11/zlibvc.vcxproj b/contrib/vstudio/vc11/zlibvc.vcxproj
index c4cffcc..e8b9705 100644
--- a/contrib/vstudio/vc11/zlibvc.vcxproj
+++ b/contrib/vstudio/vc11/zlibvc.vcxproj
@@ -239,6 +239,7 @@
       <DataExecutionPrevention>
       </DataExecutionPrevention>
       <ImportLibrary>$(OutDir)zlibwapi.lib</ImportLibrary>
+      <ImageHasSafeExceptionHandlers>false</ImageHasSafeExceptionHandlers>
     </Link>
     <PreBuildEvent>
       <Command>cd ..\..\masmx86
@@ -340,6 +341,7 @@ bld_ml32.bat</Command>
       <DataExecutionPrevention>
       </DataExecutionPrevention>
       <ImportLibrary>$(OutDir)zlibwapi.lib</ImportLibrary>
+      <ImageHasSafeExceptionHandlers>false</ImageHasSafeExceptionHandlers>
     </Link>
     <PreBuildEvent>
       <Command>cd ..\..\masmx86
